crates/shared/src/http_solver.rs

```diff
@@ -94,15 +102,15 @@ impl HttpSolverApi for DefaultHttpSolverApi {
         &self,
         model: &model::BatchAuctionModel,
         timeout: Duration,
-    ) -> Result<model::SettledBatchAuctionModel> {
+    ) -> Result<model::SettledBatchAuctionModel, Error> {
         // The timeout we give to the solver is one second less than
         // the deadline to make up for overhead from the network.
         // We use one second because the old MIP solver uses integer timeouts.
         let solver_timeout = timeout
             .checked_sub(Duration::from_secs(1))
             .context("no time left to send request")?;
 
-        let mut url = self.base.join("solve")?;
+        let mut url = self.base.join("solve").context("join base")?;
```

Comment:
This might even be `.unwrap()` worthy. AFAIK, this can only fail if `"solve"` is not a valid URL path segment.

Reply:
I thought so too but wasn't completely sure so I kept the same behavior as before.